okhttp/src/main/java/okhttp3/RequestBody.kt

```diff
@@ -107,56 +108,108 @@ abstract class RequestBody {
           charset = resolvedCharset
         }
       }
-      val bytes = content.toByteArray(charset)
-      return create(finalContentType, bytes)
+      val bytes = toByteArray(charset)
```

Comment:
We should arguably prefer forms using ByteString here over ByteArray

Reply:
This is String’s built in API. Boxing it is just an extra allocation.

Reply:
Yep, following through it appears optimal for allocations.